### PR TITLE
Maven: Add "package","library","repository" and "repo" to trigger words.

### LINKED ISSUE #3050: Maven: Add "library" "package", "repository" and "repo" to trigger words
### Description
As with, e.g. [MetaCPAN](https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/MetaCPAN.pm#L13) adding additional triggers can help surface an IA more effectively.

This should possibly trigger on more terms, e.g.:

- maven repo
- maven repository
- maven package
- java package
- ...

## People to notify
@nicoulaj 

------ 

IA Page:  https://duck.co/ia/view/maven

### PR DESCRIPTION
**Pull request by jeyadossmanimaran**
## Description: Added Additional Triggers to the maven IA

Example Query: [java package](https://www.duckduckgo.com/?q=java+package), [perl package](https://www.duckduckgo.com/?q=perl+package)

Tab Name: Software

Source: search.maven.org

image

## Related Issues and Discussions

Fixes #3050 

## People to notify
@nicoulaj 

---

https://duck.co/ia/view/maven

### CODE REVIEW

**Comment by daxtheduck** [comment]
@jeyadossmanimaran Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by jeyadossmanimaran** [comment]
Dear daxtheduck,
                          Thanks for Clarifying.But I found the above topic in the issues to be fixed and was tagged low hanging fruit and mission programming.It is because of that i added the triggers and requested a pull. I'm new to this contribution can you please elaborate the case so I can still try make some more Contribution to the case.Thank you

**Comment by jeyadossmanimaran** [comment]
I have edited the comment and have included the ia page into comment and also included the queries now.Thank you

File: lib/DDG/Spice/Maven.pm [hunk 1]
```diff
@@ -4,7 +4,7 @@ package DDG::Spice::Maven;
 use strict;
 use DDG::Spice;
 
-triggers startend => "maven", "mvn";
+triggers startend => "maven", "mvn","library","repository","repo","package";
```
Comment (tagawa) [inline lib/DDG/Spice/Maven.pm]: These triggers are a bit too general - it means that a search for `library book` would trigger this IA.

Instead, please could they be changed to `"maven library", "maven repository", "maven repo", "maven package"`? That way, only maven-related searches will trigger this IA.

**Comment by tagawa** [comment]
Thank you for this improvement @jeyadossmanimaran! It just needs a small change so I've added a comment in the code. Then we can install it on the beta server for testing.

**Comment by jeyadossmanimaran** [comment]
thank you @tagawa   for correcting and clarifying .I will look into it and change it as fast as possible

File: lib/DDG/Spice/Maven.pm [hunk 2]
```diff
@@ -4,15 +4,22 @@ package DDG::Spice::Maven;
...
 spice to => 'http://search.maven.org/solrsearch/select?q=$1&rows=5&wt=json&callback={{callback}}';
 spice wrap_jsonp_callback => 1;
 spice proxy_cache_valid   => "418 1d";
 
 handle remainder => sub {
-    return $_ if $_;
-    return;
+if($req->matched_trigger=~ m/java (library|lib|module)/ |$req->matched_trigger=~ m/(lib|library|module) java/ )
+{
+return $req->matched_trigger
```
Comment (moollaza) [inline lib/DDG/Spice/Maven.pm]: nit: can you please correct the indentation?

```perl
if ( ... ) {
    return ....;
}
elsif $_ {
    return $_;
}
return ;
```

Comment (moollaza) [inline lib/DDG/Spice/Maven.pm]: We can also use an `elsif` for that last `else` block 👍

File: lib/DDG/Spice/Maven.pm [hunk 3]
```diff
@@ -4,15 +4,22 @@ package DDG::Spice::Maven;
 use strict;
 use DDG::Spice;
 
-triggers startend => "maven", "mvn";
-
+triggers any => "maven", "mvn","repository","repo","java library","package","library java","lib java","java lib","module java","java module";
+# triggers any=> "/([a-z].+)?.?(java|maven|mvn).+?([a-z].+)?.+?(repository|repo|library|package)/ig","/([a-z].+)?.+?(repository|repo|library|package).+?([a-z].+)?.+?(java|maven|mvn)/ig";
```
Comment (moollaza) [inline lib/DDG/Spice/Maven.pm]: let's remove this 👍

File: lib/DDG/Spice/Maven.pm [hunk 4]
```diff
@@ -4,15 +4,22 @@ package DDG::Spice::Maven;
 use strict;
 use DDG::Spice;
 
-triggers startend => "maven", "mvn";
-
+triggers any => "maven", "mvn","repository","repo","java library","package","library java","lib java","java lib","module java","java module";
```
Comment (moollaza) [inline lib/DDG/Spice/Maven.pm]: @jeyadossmanimaran these are too broad. The triggers we should be added are the ones outlined in the issue:

`maven repo`, `maven repository`, `maven package`, `java package`.

We can keep `java lib` and `java module` as well

Please also update the test file with tests that ensure these work as expected 👍

**Comment by pjhampton** [comment]
ping @jeyadossmanimaran 😄

**Comment by moollaza** [comment]
**Cheat Sheet ID/Filename Convention:**
<br>- IA Page ID: `foo_bar_cheat_sheet`
<br>- JSON ID Property: `"id": "foo_bar_cheat_sheet"`
<br>- JSON Filename: `foo-bar.json`

**Comment by moollaza** [comment]
This PR is being closed due to inactivity. If you're still interested in finishing, please let me know and I'll gladly re-open it for you!

**Pull request closed without merging**